### PR TITLE
Adding error message when no flags are specified

### PR DESCRIPTION
**Pull request by Griffin-Sullivan**
Added an error message: "Could not determine an appropriate provider" when the user does not specify options to certain commands. Ex: `broker execute` or `broker checkout` with no workflow or nick.

### CODE REVIEW

**Comment by Griffin-Sullivan** [comment]
Just realized I did not test this enough. Going to update the commit with a fix.

**Comment by Griffin-Sullivan** [comment]
Ok from what I tested, this should work. Also, I'm not sure what I did to make my commits so weird, so if someone could help resolve that, it would be much appreciated.

File: broker/broker.py [hunk 1]
```diff
@@ -92,9 +92,14 @@ def __init__(self, **kwargs):
...
         # determine the provider actions based on kwarg parameters
         self._provider_actions = {}
+        no_provider = False
         for key, action in PROVIDER_ACTIONS.items():
             if key in kwargs:
                 self._provider_actions[key] = action
+            else:
+                no_provider = True
+        if no_provider:
+            logger.error("Could not determine an appropriate provider")
```
Comment (JacobCallahan) [inline broker/broker.py:102]: We can simplify further!
```suggestion
        for key, action in PROVIDER_ACTIONS.items():
            if key in kwargs:
                self._provider_actions[key] = action
        if not self._provider_actions:
            logger.error("Could not determine an appropriate provider")
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: broker/broker.py (hunk 1 above)
Comment (Griffin-Sullivan) [reply broker/broker.py:102]: Oh I see. I didn't think to check that `_provider_actions` was null that way.

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**